src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -62,6 +62,10 @@
     @JsonIgnoreProperties("exercise")
     private Set<ProgrammingExerciseTestCase> testCases = new HashSet<>();
 
+    @OneToMany(mappedBy = "exercise", cascade = CascadeType.REMOVE, orphanRemoval = true, fetch = FetchType.LAZY)
+    @JsonIgnore
+    private Set<ExerciseHint> exerciseHints = new HashSet<>();
```

Comment:
should this not be placed into Exercise?
At least on the other side `ExerciseHint` you specify the type as Exercise.
This would also make it possible to use ExerciseHints for other exercises in the future

Reply:
You are right! We should already put it into the Exercise class.